### PR TITLE
[SONARLINT] Fixing a bunch of sonarlint wanrings across the project

### PR DESCRIPTION
**Pull request by chibenwa**
- Bug fixes
 - Performance enhancements
 - Code style refactorings

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/FileIntoAction.java [hunk 1]
```diff
@@ -73,26 +73,17 @@ public void execute(Action action, Mail mail, ActionContext context) throws Mess
...
-                destinationMailbox =  HIERARCHY_DELIMITER + destinationMailbox;
-            }
-            
-            final String mailbox = destinationMailbox.replace(HIERARCHY_DELIMITER, '/');
-            final String url = "mailbox://" + recipient.asString() + mailbox;
-            //TODO: copying this message so many times seems a waste
-            context.post(url, aMail);
-            delivered = true;
-        } catch (MessagingException ex) {
-            LOGGER.debug("Error while storing mail into {}.", destinationMailbox, ex);
```
Comment (Arsnael) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/FileIntoAction.java:91]: So we removing here the debug error log statement so we can get rid of the try catch block right? Is that ok?

**Review by Arsnael — APPROVED** [review]
Nice little fixes :)

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/FileIntoAction.java (hunk 1 above)
Comment (chibenwa) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/FileIntoAction.java:91]: It's OK as the exception is re-thrown.

Log then throw is a classic anti-pattern upon exception handling.

**Comment by Arsnael** [comment]
Rebase to solve conflicts

**Pull request merged**